client/src/App.tsx

```diff
@@ -1,18 +1,57 @@
+import { useEffect, useState } from 'react';
 import { Routes, Route } from 'react-router-dom';
-import './App.css';
+import AppContext from './components/AppContext';
 import NavBar from './components/NavBar';
 import JobDetails from './pages/JobDetails';
 import Jobs from './pages/Jobs';
+import AuthPage from './pages/AuthPage';
+import './App.css';
+import { Auth, User } from './lib';
+import Home from './pages/Home';
+
+const tokenKey = 'designengineer';
 
 export default function App() {
+  const [user, setUser] = useState<User>();
+  const [token, setToken] = useState<string>();
+  const [isAuthorizing, setIsAuthorizing] = useState(true);
+
+  useEffect(() => {
+    const auth = localStorage.getItem(tokenKey);
+    if (auth) {
+      const a = JSON.parse(auth);
+      setUser(a.user);
+      setToken(a.token);
+    }
+    setIsAuthorizing(false);
+  }, []);
+
+  if (isAuthorizing) return null;
+
+  function handleSignIn(auth: Auth) {
+    localStorage.setItem(tokenKey, JSON.stringify(auth));
```

Comment:
Nice! Most students just put the token in local storage and then realize they have trouble authorizing again without asking the user to login. I always tell them to put the entire auth into local storage. You've already done that!